controllers/SpotifyTrackController.js

```diff
@@ -1,288 +1,183 @@
-                    body.items[i].track.artists.forEach(element => {
-                        temp_artist.push(element.name);
-                    });
-                    const newTrack ={
-                        title: body.items[i].track.name,
-                        artist: temp_artist,
-                        _id: body.items[i].track.id,
-                        prewiewURL: body.items[i].track.preview_url,
-                        uri: body.items[i].track.uri
-                    };
-
-                    //#region aktualizowanie jeśli znajdzie, a jeśli nie ma to dodaje nowy dokument
-                    await TrackModel.findOneAndUpdate({ _id: newTrack._id }, newTrack, { upsert: true })
-                        .then(() => {
-                            found++;
-                        })
-                        .catch(err => {
-                            console.log(err.message);
-                        });
-                    // #endregion
-                }
-                console.log(`Found ${found} tracks.`);
-                // #endregion
-
-                // #region dodawanie pojedynczej piosenki bez sprawdzenia czy istnieje
-                // let temp_artist =[];
-                // body.artists.forEach(element => {
-                //     temp_artist.push(element.name);
-                // });
-
-                // const newTrack = new TrackModel({
-                //     title: body.name,
-                //     artist: temp_artist,
-                //     _id: body.id,
-                //     prewiewURL: body.preview_url,
-                //     uri: body.uri
-                // });
-                // console.log(newTrack);
-
-                // newTrack.save()
-                //     .then(() => {
-                //         res.redirect('/');
-                //     })
-                //     .catch((err) => {
-                //         console.log(err);
-                //     })
-                // #endregion
-
-                res.redirect('/adminPanel');
-            });
+
+    request.get(options, async (error, response, body) => {
+        let found = 0
+
+        for(let i = 0; i < body.total; i++) {
+            if (body.items[i].track.explicit) continue
+
+            let temp_artist = []
+            body.items[i].track.artists.forEach(element => {
+                temp_artist.push(element.name);
+            })
+
+            const newTrack = {
+                title: body.items[i].track.name,
+                artist: temp_artist,
+                _id: body.items[i].track.id,
+                prewiewURL: body.items[i].track.preview_url,
+                uri: body.items[i].track.uri
+            }
+
+            await TrackModel.findOneAndUpdate({ _id: newTrack._id }, newTrack, { upsert: true })
+                .catch(err => {
+                    console.log(err.message)
+                })
+
+            found++
         }
-    });
-};
+
+        console.log(`Found ${found} tracks.`)
```

Comment:
Linijka `console.log('Found ${found} tracks.')` jak i zmienna '_**found**_' później zniknie, tak?
 Na czas pracy może zostać, tylko nie bardzo wiem jaki jest sens zostawienia tego w końcowym etapie.
 Taka mała uwaga tylko.

Reply:
Found zostaje ew. zostanie zmieniona nazwa. To jest prawie jedyne miejsce, gdzie programista wie co się dzieje na stronie. Drugim miejscem jest rządania tokena przy logowaniu to spotify api.